src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -313,6 +321,16 @@ private static void GetGenericInfo(PEModuleSymbol moduleSymbol, TypeDefinitionHa
                 mangleName = !ReferenceEquals(_name, metadataName);
             }
 
+            // when a file-local type from source is loaded from metadata, we do a best-effort check to identify it as a file type
+            // this is needed to allow EE to bind to file types from metadata, for example.
+            if (container.IsNamespace && GeneratedNameParser.TryParseFileTypeName(_name, out var displayFileName, out var ordinal, out var originalTypeName))
```

Comment:
>if (container.IsNamespace && GeneratedNameParser.TryParseFileTypeName(_name, out var displayFileName, out var ordinal, out var originalTypeName)) [](http://example.com/codeflow?start=12&length=145)

To clarify my previous comment, here, before this ```if``` I suggest to throw unreachable if ```_lazyUncommonProperties``` is not null. Inside the ```if``` I suggest assign directly to ```_lazyUncommonProperties```.